source/codegen/templates/service.cpp.mako

```diff
@@ -37,6 +37,14 @@ namespace ${config["namespace_component"]}_grpc {
   {
   }
 
+% if common_helpers.has_viboolean_array_param(functions):
+  void ${service_class_prefix}Service::Copy(const std::vector<ViBoolean>& input, google::protobuf::RepeatedField<bool>* output) 
```

Comment:
I couldn't find a way to make this method generic. If we did make it generic, callers will have to pass ViBoolean and bool as generic type and possibly lambda for converting from input type to output type? This is also the reason I couldn't change the check from `parameter['type'] == 'ViBoolean[]'` to `is_non32bit_integer` as no generic Copy method could be implemented. Is this fine?

Reply:
Bool is special because you don't generally want to do `(bool)item` to go from `ViBoolean` to `bool`. That would be the right approach for other non-32-bit numeric though. But here's a way to make this generic (i haven't tested it):
```cpp
template<class TIn, class TOut, class TConvert>
void ${service_class_prefix}Service::Copy(const std::vector<TIn>& input, google::protobuf::RepeatedField<TOut>* output, TConvert converter)
{
  for (auto item : input) {
    output->Add(converter(item));
  }
}
```
Then you could pass a lambda with the custom boolean logic as the 3rd parameter.
I think what you have here for `ViBoolean` is okay for this PR. Adding a 3rd parameter would affect the codegen logic in the caller. I'd say let's handle that when we need it.

Reply:
Yep, makes sense. Leaving it as is for now.